openzen_sensor: 1.0.0-1 in 'melodic/distribution.yaml' [bloom]

th-lp opened the pull request:
Increasing version of package(s) in repository `openzen_sensor` to `1.0.0-1`:

- upstream repository: https://bitbucket.org/lpresearch/openzenros.git
- release repository: https://github.com/lp-research/openzen_sensor-release.git
- distro file: `melodic/distribution.yaml`
- bloom version: `0.9.3`
- previous version for package: `null`

## openzen_sensor

```
* Initial Release of OpenZen ROS driver
* Contributors: LP-Research Inc. Team
```

th-lp committed:
openzen_sensor: 1.0.0-1 in 'melodic/distribution.yaml' [bloom]

th-lp commented:
Some of the integration builds are failing because of this error:
"Pull Request builds only supported on GitHub right now. Cannot do pull request against bitbucket.org"
Our upstream repository is hosted on bitbucket.org. Is there still a way to test this Pull request? Thanks.

clalancette commented:
> Some of the integration builds are failing because of this error:
> "Pull Request builds only supported on GitHub right now. Cannot do pull request against bitbucket.org"
> Our upstream repository is hosted on bitbucket.org. Is there still a way to test this Pull request? Thanks.

I just want to be clear that there are two different pull request testing that are being talked about here.  For this PR in this repository, the pull request testing will always fire and always check that the additions fit our formatting, style, etc.  No work needs to be done by you, other than making sure these pass.

The pull request testing that the nosetests are complaining about have to do with testing pull requests *to your source repository*.  In order to do that, specific hooks need to be setup on your repository/organization.  Unfortunately, we only support those hooks for Github (and maybe Gitlab, though I'm less sure on that one).  So the short is that no, we can't do pull request testing for you from bitbucket.

That being said, all other parts of releasing a package into ROS should work from bitbucket.  In order to get this to pass, you'll have to remove the `test_pull_requests: true` line from this PR, and things should start passing.

th-lp committed:
removed testing of pull-request because this is not supported for non-github upstream repositories

th-lp committed:
switched to master branch for src and doc sections

th-lp commented:
@clalancette thanks for the detailed explanations! I removed the `test_pull_requests:true` line and switched the `version:` entry in the `distribution.yaml` file to the master branch. Before it was errouneously pointing to a fixed git tag.

tfoote reviewed (changes requested):
This package lists MIT as the license, but it has not LICENSE file and no licence headers in the source. 

And the submodule that is included has a LICENSE file with LGPL inside it that's not represented in the package.xml too. Please make sure to list all licenses in the packages.xml and make sure that there is proper documentation of the license outside the package.xml

th-lp commented:
@tfoote Thanks for pointing out the problem with the License documentation. I added the LICENSE file to the repository and extended the list of licenses in the package.xml file. Should I create a new pull-request or is there a way to update our release repository using bloom and keep the release 1.0.0 number and this pull-request?

I have an additional question concerning the licensing topic:
This folder in our repository https://bitbucket.org/lpresearch/openzen/src/master/external/ contains three proprietary Windows DLLs for interfacing to IO chips. They have non-open source proprietary licenses and the vendors allow the distribution of these driver DLLs for the usage with the hardware of the respective vendor. No source code of these drivers is compiled in the ROS compile and the DLL are not copied to the installation directory of the ROS package. Is it in compliance with ROS license regulations to have them in a cloned repository? If this is a problem, I can move them to an outside repository.

Thanks for your help!

th-lp commented:
@tfoote and @clalancette can you have a look at my recent changes and question above? Thanks.

clalancette commented:
> I have an additional question concerning the licensing topic:
> This folder in our repository https://bitbucket.org/lpresearch/openzen/src/master/external/ contains three proprietary Windows DLLs for interfacing to IO chips. They have non-open source proprietary licenses and the vendors allow the distribution of these driver DLLs for the usage with the hardware of the respective vendor. No source code of these drivers is compiled in the ROS compile and the DLL are not copied to the installation directory of the ROS package. Is it in compliance with ROS license regulations to have them in a cloned repository? If this is a problem, I can move them to an outside repository.

It seems fine to me, especially since they aren't installed as part of the debian package.  @tfoote , thoughts?

tfoote commented:
I looked through the first license I found at https://bitbucket.org/lpresearch/openzen/src/master/external/ftdi/LICENSE and it doesn't give the right to redistribute. And the software is stated to also be under an additional license linked to at: http://www.ftdichip.com/FTDriverLicenceTerms.htm but that is no longer a valid link. 

The distinction of if something is compiled into the package and installed is the most important for us and our debian packages. However any users who pull from source will be unhappy to find things in their source tree that are not under the declared licenses in the package.xml And if they were to redistribute in a way that's not our formal installation rule but to tar up a workspace these elements would all be included. It would be much cleaner to setup openzen as a dependency that can be installed separately.

We're not super strict on what exactly you can do or not do, but we do want things to be clear what people are getting from the package so I'd say it can stay if you extend the package.xml to include the note about the embedded proprietary binaries.

ThomasHauth commented:
Thanks for your feedback. I will remove the proprietary files from the repository and update this pull-request.

hidmic commented:
@th-lp friendly ping.

th-lp commented:
Hi @hidmic Thanks for the ping. I am still working on the PR update and I will have it ready by tomorrow.

th-lp commented:
Hi @hidmic Thanks for the ping. I am currently working on the revised PR and will have it ready by tomorrow.

th-lp committed:
Increased to newer openzen_sensor version which does not include any binary files any more and only uses open-source licenses.

th-lp commented:
@hidmic @tfoote 
I bumped the openzen_sensor version to 1.0.1 which contains modifications to the OpenZen ROS repository as follows:

1) Binary proprietary device driver DLLs have been moved to a separate repository and are downloaded via CMake FetchContent if required. For the ROS build, this download is disabled. Now only open-source licenced libraries are contained in the build folder for the ROS build. These licenses are MIT, Boost Software License, LGPLv3, BSD and are all listed via the <license> tags of the package.xml file.

See line 11 and 12 in this file for the binary download options and that they are disabled by default:
https://bitbucket.org/lpresearch/openzenros/src/cb9d2bdcdfca3a64b61b13d40b41b659f736d026/CMakeLists.txt

2) Switched to a newer version with OpenZen 1.1.3 which contains some bugfixes and adds support for one new sensor model. See https://bitbucket.org/lpresearch/openzen/src/master/CHANGELOG.md for details.

Please let me know if these changes comply better with the ROS license rules. Thanks for your help and suggestions!

ivanpauno reviewed (approved):
LGTM!
We're holding melodic packages until the next sync is done (see [here](https://discourse.ros.org/t/preparing-for-melodic-sync-2020-09-10/16350)), this will be merged after that happens.

tfoote reviewed (approved):
Thank you for separating out the content.
